### PR TITLE
Delete clusterRole *aggregate* when RolloutManager is deleted

### LINKED ISSUE #85: ClusterRoles '*aggregate*'  should be deleted when RolloutManager is deleted
ClusterRoles argo-rollouts-aggregate-to-admin, argo-rollouts-aggregate-to-edit and argo-rollouts-aggregate-to-view do not get deleted upon deletion of Rolloutmanager instance.

### To reproduce
- Install the operator
- Create a RolloutManager instance
- Delete the instance and check if the relevant clusterroles are deleted 

### Actual Results 

ClusterRoles argo-rollouts-aggregate-to-admin, argo-rollouts-aggregate-to-edit and argo-rollouts-aggregate-to-view do not get deleted

### Expected results:

ClusterRoles argo-rollouts-aggregate-to-admin, argo-rollouts-aggregate-to-edit and argo-rollouts-aggregate-to-view should get deleted

### Fix Criteria
- The above cluster roles/bindings should be deleted, similar to how we are already deleting the other CR/CRB that we create (See this PR https://github.com/argoproj-labs/argo-rollouts-manager/pull/69)
- E2E/unit test to verify the expected behaviour

### PR DESCRIPTION
**Pull request by Rizwana777**
JIRA - https://issues.redhat.com/browse/GITOPS-5388

Fixes #85

### CODE REVIEW

File: controllers/utils.go [hunk 1]
```diff
@@ -232,7 +232,7 @@ func validateRolloutsScope(cr rolloutsmanagerv1alpha1.RolloutManager, namespaceS
 			return &reconcileStatusResult{
 				rolloutController: &phaseFailure,
 				phase:             &phaseFailure,
-			}, fmt.Errorf(UnsupportedRolloutManagerNamespaceScoped)
+			}, errors.New(UnsupportedRolloutManagerNamespaceScoped)
```
Comment (chetan-rns) [inline controllers/utils.go:235]: The changes in this file are already present in the main branch 🤔  Can you try to rebase again?

File: tests/e2e/cluster-scoped/cluster_scoped_rollouts_test.go [hunk 2]
```diff
@@ -324,5 +325,53 @@ var _ = Describe("Cluster-scoped RolloutManager tests", func() {
...
+
+			clusterRoleView := &rbacv1.ClusterRole{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "argo-rollouts-aggregate-to-view",
+				},
+			}
+
+			Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(clusterRoleAdmin), clusterRoleView)).To(Succeed())
+			Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(clusterRoleEdit), clusterRoleView)).To(Succeed())
+			Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(clusterRoleView), clusterRoleView)).To(Succeed())
```
Comment (chetan-rns) [inline tests/e2e/cluster-scoped/cluster_scoped_rollouts_test.go:357]:
```suggestion
			Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(clusterRoleAdmin), clusterRoleAdmin )).To(Succeed())
			Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(clusterRoleEdit), clusterRoleEdit)).To(Succeed())
			Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(clusterRoleView), clusterRoleView)).To(Succeed())
```

File: tests/e2e/cluster-scoped/cluster_scoped_rollouts_test.go [hunk 3]
```diff
@@ -324,5 +325,53 @@ var _ = Describe("Cluster-scoped RolloutManager tests", func() {
...
+				return k8sClient.Get(ctx, client.ObjectKeyFromObject(clusterRoleAdmin), clusterRoleView)
+			}, "1m", "1s").ShouldNot(Succeed())
+
+			Eventually(func() error {
+				return k8sClient.Get(ctx, client.ObjectKeyFromObject(clusterRoleView), clusterRoleView)
+			}, "1m", "1s").ShouldNot(Succeed())
+
+			Eventually(func() error {
+				return k8sClient.Get(ctx, client.ObjectKeyFromObject(clusterRoleEdit), clusterRoleView)
+			}, "1m", "1s").ShouldNot(Succeed())
```
Comment (chetan-rns) [inline tests/e2e/cluster-scoped/cluster_scoped_rollouts_test.go:373]:
```suggestion
			Eventually(func() error {
				return k8sClient.Get(ctx, client.ObjectKeyFromObject(clusterRoleAdmin), clusterRoleAdmin)
			}, "1m", "1s").ShouldNot(Succeed())

			Eventually(func() error {
				return k8sClient.Get(ctx, client.ObjectKeyFromObject(clusterRoleView), clusterRoleView)
			}, "1m", "1s").ShouldNot(Succeed())

			Eventually(func() error {
				return k8sClient.Get(ctx, client.ObjectKeyFromObject(clusterRoleEdit), clusterRoleEdit)
			}, "1m", "1s").ShouldNot(Succeed())
```

File: tests/e2e/cluster-scoped/cluster_scoped_rollouts_test.go [hunk 4]
```diff
@@ -324,5 +325,53 @@ var _ = Describe("Cluster-scoped RolloutManager tests", func() {
...
+			Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(clusterRoleEdit), clusterRoleView)).To(Succeed())
+			Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(clusterRoleView), clusterRoleView)).To(Succeed())
+
+			By("Delete RolloutManager")
+			Expect(k8sClient.Delete(ctx, &rolloutsManagerCl)).To(Succeed())
+
+			By("Verify clusterRole '*aggregate*' is deleted")
+			Eventually(func() error {
+				return k8sClient.Get(ctx, client.ObjectKeyFromObject(clusterRoleAdmin), clusterRoleView)
+			}, "1m", "1s").ShouldNot(Succeed())
```
Comment (chetan-rns) [inline tests/e2e/cluster-scoped/cluster_scoped_rollouts_test.go:365]: I think `ExistByName` fixture would be more appropriate to use here and in the Expect statements above.

```
Eventually(clusterRoleAdmin, "1m", "1s").ShouldNot((k8s.ExistbyName(k8sClient)))
```

File: controllers/utils.go (hunk 1 above)
Comment (Rizwana777) [reply controllers/utils.go]: Yes, When I raised this PR, it was not there, rebased it now

**Review by jgwest — APPROVED** [review]
LGTM, thanks @Rizwana777, and thanks @chetan-rns  for reviewing!

**Pull request merged**